get_ending_subscriptions.py

```diff
@@ -0,0 +1,112 @@
+    else:
+        end_date = datetime.now() + timedelta(days=1)
+    #
+    messageset = MessageSet.objects.get(short_name=messageset_name)
+    set_max = messageset.messages.filter(lang="eng_ZA").count()
+    schedule = messageset.default_schedule
+    #
+    # Number of times the schedule will run before the start of the timeframe
+    runs_before_start = len(
+        schedule.get_run_times_between(datetime.now(), start_date))
+    # Number of times the schedule will run before the end of the timeframe
+    runs_before_end = len(
+        schedule.get_run_times_between(datetime.now(), end_date))
+    #
+    # Any schedules older than this will expire before the window
+    max_next_msg = set_max - runs_before_start
+    # Any schedules newer than this will expire after the window
+    min_next_msg = set_max - runs_before_end
+    #
+    subscriptions = Subscription.objects.filter(
+        messageset=messageset, active=True, completed=False,
+        next_sequence_number__gt=min_next_msg,
+        next_sequence_number__lte=max_next_msg)
+    #
+    identity_client = IdentityStoreApiClient(
+        identity_store_token, identity_store_url)
+    hub_client = HubApiClient(hub_token, hub_url)
+    #
+    compile_temp_csv(subscriptions, hub_client, identity_client)
+
+
+def compile_temp_csv(subscriptions, hub_client, identity_client):
+    print("Retrieving and writing data to terminating_subscriptions.csv")
+    with open("terminating_subscriptions.csv", 'w+b') as csvfile:
+        fieldnames = [
+            'Subscription', 'MSISDN', 'Facility code', 'EDD', 'ID Number',
+            'Language']
+        csvwriter = csv.DictWriter(csvfile, fieldnames=fieldnames,
+                                   restval="Unavailable")
+        csvwriter.writeheader()
+        for sub in subscriptions.iterator():
+            row_data = {'Subscription': sub.id}
+            print("Retrieving registration data for %s" % sub.id)
+            registrations = hub_client.get_registrations(
+                {'registrant_id': sub.identity})
+            if registrations['count']:
+                try:
+                    row_data.update(
+                        get_registration_data(registrations['results'][0]))
+                except ConnectionError:
+                    print("Error retrieving registration data.")
+            if "MSISDN" not in row_data or "EDD" not in row_data or "Language"\
+                    not in row_data:
+                print("Some registration data missing. Retrieving identity "
+                      "data for %s" % sub.id)
+                try:
+                    identity = identity_client.get_identity(sub.identity)
+                    if identity:
+                        row_data.update(get_identity_data(identity))
+                except ConnectionError:
+                    print("Error retrieving identity data.")
+            csvwriter.writerow(row_data)
+        csvfile.seek(0)
```

Comment:
Is there a reason to seek if we're not doing anything with the file from here on?